tests/build_tool/test_build_tool.py

```diff
@@ -41,6 +43,35 @@ def test_idempotency(build_tool_validator, tmp_path, test_case_dir):
     build_tool_validator.assert_same_config_dirs(outputdir, sample_test_case / "expected")
 
 
+def test_failure(build_tool_validator, tmp_path, fail_test_case):
+    outputdir = tmp_path / "output"
+    custom_schemadir = fail_test_case / "schemas"
+    schemadir = (
+        custom_schemadir if custom_schemadir.exists() else pathlib.Path("schemas").absolute()
```

Comment:
When I read the successful test case I do not see `--schemadir` being passed. I assume because it uses some default, what is the intent here when the custom is not present. I would use the same approach if both cases, either provide `--schemadir` to success and failure, or do not provide `--schemadir` to failure test cases for which the custom_schemadir is not present.

Reply:
I didn't have a use case for happy day tests to support a custom set of schemas, so they use the "production ones" in `./schemas` relative to `build.py`. That also indirectly tests that the build tool would work without the `--schemadir` option.

There's certainly room for improvement. This PR might not be the best place to try to fix everything about the tests though.